### ISSUE TITLE
ViT and 3D images

### ISSUE THREAD

**Issue by vlatorre847** [lucidrains/vit-pytorch#125]
Hi,

I would like to use the ViT to solve a regression problem with three dimensional images.

Did anyone already try to do such a thing?

V. L.

**Comment by lucidrains** [comment]
@vlatorre847 yes, attention is all you need. just do the patching to tokens, but take 3 dimensional patches instead

**Comment by cmartin-isla** [comment]
Hello, thanks for this nice repo. I have a question. I work normally with grayscale images and not quite familiar with einops. Does that imply that chanels are simply concatenated? here for example we would have a concatenation of unzipped 3d patches, c times?

**Comment by xiaoyu0318** [comment]
> @vlatorre847 yes, attention is all you need. just do the patching to tokens, but take 3 dimensional patches instead
> 
> ```python
> x = rearrange(x, 'b c (x p1) (y p2) (z p3) -> b (x y z) (c p1 p2 p3)', p1 = p, p2 = p, p3 = p)
> ```

how to invert to original order?
